packages/flutter_tools/lib/src/plugins.dart

```diff
@@ -466,10 +469,15 @@ const String _objcPluginRegistryImplementationTemplate = '''//
 //
 
 #import "GeneratedPluginRegistrant.h"
+
 {{#plugins}}
+#if __has_include(<{{name}}/{{class}}.h>)
```

Comment:
Swift plugins should not need to add an Objective-C class if they don't need it.